### PR TITLE
[DEV only, test purpose, do not merge ][REFACTOR] Menu component

### PR DESCRIPTION
**Pull request by panaC**
refactor menu : 
- children props
- MenuButton and MenuContent component merged in Menu
- doBackFocusMenuButton logic extracted from atomic component and handled in redux-saga on modal closing
- create a focusRef store injected in DI and React-Context
- https://floating-ui.com/docs/getting-started on menuContent

### CODE REVIEW

**Comment by panaC** [comment]
image
image

**Comment by panaC** [comment]
@arthur-lemeur I have a problem with the z-index of arrow div. How to fix it ?

File: src/renderer/common/components/menu/MenuContent.tsx [hunk 1]
```diff
@@ -29,55 +27,119 @@ interface IBaseProps {
...
+                        left: "right", // not used
+                    }[placement.split("-")[0]];
+
+                    this.setState({
+                        menuStyle: {
+                            content: {
+                                position: "absolute",
+                                left: `${x}px`,
+                                top: `${y}px`,
+                                zIndex: "1",
```
Comment (arthur-lemeur) [inline src/renderer/common/components/menu/MenuContent.tsx:126]: z-index was missing here. That solved the problem of the arrow not showing in the list view.

**Comment by panaC** [comment]
defined #2016

**Pull request closed without merging**